docs/versioned_docs/version-v1.7.0/features/secrets/general.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Consuming Secrets"
+layout: default
+---
+
+# Consuming secrets
+Every provider implements `ISecretProvider` which makes it easy to use a consistent flow, regardless of the provider.
+
+Secrets can be easily retrieved as follows:
+
+```csharp
+Secret secret = await secretProvider.GetSecretAsync("EventGrid-AuthKey");
+
+string secretValue = secret.Value;
+string secretVersion = secret.Version;
+DateTimeOffset? expirationDate = secret.Expires;
+```
+
+# Raw secrets
+In some scenarios you'd like to just get the secret value directly without any metadata.
+This is possible by calling the `...Raw...` variants on the `ISecretProvider` implementations.
+
+```csharp
+string secretValue = await secretProvider.GetRawSecretAsync("EventGrid-AuthKey");
+```
+
+# Caching Secrets
+Some secret providers recommend to cache secrets for a while to avoid hitting the service limitations.
+
+We provide a `CachedSecretProvider` which allows the secrets to be cached in memory for a certain amount of time.
+
+```csharp
+var cachedSecretProvider = new CachedSecretProvider(secretProvider);
+Secret secret = await cachedSecretProvider.GetSecretAsync("EventGrid-AuthKey");
+```
+
+If you prefer a more fluent approach you can also use our `WithCaching` extension.
+
+```csharp
+var cachedSecretProvider = new KeyVaultSecretProvider(vaultAuthentication, vaultConfiguration)
+                                    .WithCaching();
+Secret secret = await cachedSecretProvider.GetSecretAsync("EventGrid-AuthKey");
+```
+
+## Configuring the cache
+By default, retrieved secrets are cached for **5 minutes**, but you can configure this yourself.
+
+```csharp
+var cacheConfiguration = new CacheConfiguration(TimeSpan.FromMinutes(10)); // Optional: Default is 5 min
+var cachedSecretProvider = new CachedSecretProvider(secretProvider, cacheConfiguration);
+Secret secret = await cachedSecretProvider.GetSecretAsync("EventGrid-AuthKey");
+```
+
+## Forcing a secret refresh
+In some scenarios you'd like to skip the cache and do a hard refresh by looking it up in the provider.
+
+This is important because in certain scenarios your secrets can be rolled and thus you will be revoked access.
+After a hard refresh you can use the latest secret again and proceed your work.
```

Comment:
When reading this sentence, I'd assume that you need to call a specific function to do a 'hard refresh', but in fact, it is just calling the `GetSecretAsync` method with the `ignoreCache` parameter set to true.

Therefore, I'd change this sentence to something like:
> When you want to bypass the cached secret, call the `GetSecretAsync` method and specify `true` for the `ignoreCache` parameter.

Then I also wonder, does this parameter have a side-effect of invalidating the cache for this specific secret ?

Reply:
We have a 'InvalidateSecretAsync' method on our cached secret providers that will do the 'hard refresh' that is described here.

Reply:
> Then I also wonder, does this parameter have a side-effect of invalidating the cache for this specific secret ?

This is a parameter on an interface that is publicly available to be used alongside the existing set of secret providers, so consumers can have this done by their own choosing. We do not, in fact, cause side-effects. We only by-pass the caching. https://github.com/arcus-azure/arcus.security/blob/aba283e79ce68408752a1c7ad3b0ec0db43ec979/src/Arcus.Security.Core/Caching/CachedSecretProvider.cs#L144

Reply:
I would just make this more clear, because the code sample does not align with 'forcing a a secret refresh'.
Changing the title to 'bypassing cached secrets' would improve the docs a lot imho.

Reply:
I would just make this more clear, because the code sample does not align with 'forcing a a secret refresh'.
Changing the title to 'bypassing cached secrets' would improve the docs a lot imho.

Reply:
Ah, yes, but we have to change this in all versions, then.

Reply:
Do you see a problem in that ?

Reply:
No, only will take a bit of time. Will do it probably later, after all the other more critical stuff is done. So, not in this PR.